napari/_tests/test_with_screenshot.py

```diff
@@ -10,6 +10,14 @@
 )
 
 
+@pytest.fixture
+def qt_viewer(qt_viewer_):
+    # show the qt_viewer and hide its welcome widget
+    qt_viewer_.show()
+    qt_viewer_.set_welcome_visible(False)
+    return qt_viewer_
```

Comment:
```suggestion
    yield qt_viewer_
    qt_viewer_.hide()
```

Reply:
@Czaki seems like with this change the test started raising a `RuntimeError` with PyQt6. Checking the `qt_viewer_` fixture, seems like a cleaning function is being passed to the `qtbot.addWidget` call so maybe doing the `yield` + `hide()` call is not necessary? 🤔 :

https://github.com/napari/napari/blob/b9dc8dca1e1afa7567586e7ffa7a36cc7cd9fb6f/napari/conftest.py#L402

Let me know!

Reply:
You are correct. May you put proper comment in https://github.com/napari/docs/pull/505 to clarify that hide is not neccesary?

Reply:
Seems like https://github.com/napari/docs/pull/505 has been already merged but could it be worthy then to create a new PR adding around https://github.com/napari/docs/blob/b9e26a385af6c728ca621f42e0292524281a9737/docs/developers/contributing/testing.md?plain=1#L290 something like:

```
including hidding and clearing any references to viewer instances via a cleaning function passed to the `qtbot.addWidget` as the `before_close_func` kwarg
```

For the moment will revert the suggestion but let me know if that change/addition over the docs makes sense!